TSTutorial/homeworks/hw_3/task_2.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * 
+ * Создайте дженерик класс Storage<T>, где T должен быть ограничен объектом, имеющим КАК МИНИМУМ {id: number}.
+Задача класса - хранить объекты типа Т в приватном массиве
+Реализуйте в классе следующие методы:
+  - constructor должен принимать необзятельный массив объектов соответствующего типа. 
+    Если массив пришел - объекты запушить в хранилище.
+  - add, принимающий либо объект типа Т, либо объект типа Т без id. Метод должен быть реализовать с помощью ПЕРЕГРУЗКИ.
+    Если на вход подан объект без айди - айди надо сгенерировать, затем запушить обьект в хранилище
+    Если на вход подан объект с айди - запушить его в хранилище
+    Для типизации используйте Utility Types
+  - update, принимающий объект с айди и любым набором остальных ключей из типа Т. 
+  - remove, принимающий на вход id и удаляющий объект из массива
+  - getById(id), возвращающий объект по айди если найден
+  - getAll(), возвращает все объекты в хранилище
+
+ */
+  class StorageClass<T extends { id: number }> {
+    private storage: T[] = [];
+    constructor(storage?: T[]) {
+        if (storage) this.storage = storage.map(item => item as T);
+    }
+    add(item: T): this;
+    add(item: Omit<T, "id">): this;
+    add(item: Omit<T, "id"> | T): this {
+        if ("id" in item) 
+            this.storage.push(item as T) 
+        else {
+            const entity = {id : this.generateId(), ...item} as T;
+            this.storage.push(entity);
+        }
+        return this;
+    }
+
+    update(id: number, item: T) {
```

Comment:
```suggestion
    update(id: number, item: Partial<T>) {
```
Метод должен уметь принимать любой набор ключей и значений из Т, а не целиком Т, т.е. минимум это { id: numer }